pkg/acsengine/engine.go

```diff
@@ -2854,10 +2854,10 @@ func internalGetPoolLinkedTemplateText(extTargetVMNamePrefix, orchestratorType,
 	if e != nil {
 		return "", e
 	}
-	if strings.Contains(extTargetVMNamePrefix, "agent") {
```

Comment:
agent VMs do not contain "agent", only masters contain "master". this was causing the extension vm type to always be master.